### PR TITLE
DOC: Update release process

### PR DESCRIPTION
**Pull request by datapythonista**
@lithomas1 can you have a look and let me know if everything is clear please?

### CODE REVIEW

File: doc/source/development/maintaining.rst [hunk 1]
```diff
@@ -468,8 +474,10 @@ which will be triggered when the tag is pushed.
 Post-Release
 ````````````
 
-1. Update symlink to stable documentation by logging in to our web server, and
-   editing ``/var/www/html/pandas-docs/stable`` to point to ``version/<latest-version>``.
+1. Update symlinks to stable documentation by logging in to our web server, and
+   editing ``/var/www/html/pandas-docs/stable`` to point to ``version/<latest-version>``
+   for major and minor releases, or ``version/<minor>`` to ``version/<patch>`` for
+   patch releases.
```
Comment (lithomas1) [inline doc/source/development/maintaining.rst:480]: Is there a command/script to do this?

I haven't really done symlinks before, so this is probably something that I shouldn't just figure out on the fly.

**Review by lithomas1 — COMMENTED** [review]
Thanks for writing this up. Just one minor nit.

File: doc/source/development/maintaining.rst (hunk 1 above)
Comment (datapythonista) [reply doc/source/development/maintaining.rst:480]: It's just something like `ln -sf 2.0.2 2.0` from the command line. I don't think  it's worth having a script, happy to have a call to show you, and do it together during the release (that part and any really).

Comment (datapythonista) [reply doc/source/development/maintaining.rst]: I just updated the instructions, let me know if there is anything that is not clear or can be improved.

**Review by lithomas1 — APPROVED** [review]
LGTM.

**Comment by lithomas1** [comment]
thanks @datapythonista.

**Pull request merged**